metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -0,0 +1,347 @@
+        columns.append(dbtCol)
+    return columns
+
+
+def extract_dbt_entities(
+    nodes, catalog, platform: str, environment: str
+) -> List[DBTNode]:
+    dbt_entities = []
+
+    for key in nodes:
+        node = nodes[key]
+        dbtNode = DBTNode()
+
+        dbtNode.dbt_name = key
+        dbtNode.node_type = node["resource_type"]
+        dbtNode.name = node["name"]
+        dbtNode.database = node["database"]
+        dbtNode.schema = node["schema"]
+        dbtNode.dbt_file_path = node["original_file_path"]
+
+        if "materialized" in node["config"].keys():
+            # It's a model
+            dbtNode.materialization = node["config"]["materialized"]
+            dbtNode.upstream_urns = get_upstreams(
+                node["depends_on"]["nodes"], nodes, platform, environment
+            )
+        else:
+            # It's a source
+            dbtNode.materialization = catalog[key]["metadata"]["type"]
+            dbtNode.upstream_urns = []
+
+        if dbtNode.materialization != "ephemeral":
+            dbtNode.columns = get_columns(catalog[dbtNode.dbt_name])
+        else:
+            dbtNode.columns = []
+
+        dbtNode.datahub_urn = get_urn_from_dbtNode(
+            dbtNode.database,
+            dbtNode.schema,
+            dbtNode.name,
+            platform,
+            environment,
+        )
+
+        dbt_entities.append(dbtNode)
+
+    return dbt_entities
+
+
+def loadManifestAndCatalog(
+    manifest_path: str, catalog_path: str, platform: str, environment: str
+) -> List[DBTNode]:
+
+    with open(manifest_path, "r") as manifest:
+        with open(catalog_path, "r") as catalog:
+            dbt_manifest_json = json.load(manifest)
+            dbt_catalog_json = json.load(catalog)
+
+            manifest_nodes = dbt_manifest_json["nodes"]
+            manifest_sources = dbt_manifest_json["sources"]
+
+            all_manifest_entities = manifest_nodes | manifest_sources
+
+            catalog_nodes = dbt_catalog_json["nodes"]
+            catalog_sources = dbt_catalog_json["sources"]
+
+            all_catalog_entities = catalog_nodes | catalog_sources
+
+            logger.info(
+                "loading {} nodes, {} sources".format(
+                    len(all_manifest_entities), len(all_catalog_entities)
+                )
+            )
+
+            nodes = extract_dbt_entities(
+                all_manifest_entities, all_catalog_entities, platform, environment
+            )
+
+            return nodes
+
+
+def get_urn_from_dbtNode(
+    database: str, schema: str, name: str, platform: str, env: str
+) -> str:
+    db_fqn = f"{database}.{schema}.{name}".replace('"', "")
+    return f"urn:li:dataset:(urn:li:dataPlatform:{platform},{db_fqn},{env})"
+
+
+def get_custom_properties(node: DBTNode) -> Dict[str, str]:
+    tags = {}
+    tags["dbt_node_type"] = node.node_type
+    tags["materialization"] = node.materialization
+    tags["dbt_file_path"] = node.dbt_file_path
+    return tags
```

Comment:
nice!

Reply:
I guess these are actually being loaded as properties, not tags.  I'll change the variable names to avoid confusion.